logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/CryptoTransactionProvider.kt

```diff
@@ -80,8 +81,17 @@ class CryptoTransactionProviderImpl(
     ): Either<CoreFailure, R> {
         return proteusClientProvider.getOrError()
             .flatMap { proteus ->
-                proteus.transaction(name.toTransactionName("proteus")) { proteusContext ->
-                    block(proteusContext)
+                try {
+                    proteus.transaction(name.toTransactionName("proteus")) { proteusContext ->
+                        block(proteusContext)
+                    }
+                } catch (e: IllegalStateException) {
+                    if (e.message?.contains("already been destroyed") == true) {
```

Comment:
Question: Is there an error code or something that can be caught instead of the string in the stacktrace ?

Reply:
No, it's not an core crypto error 🫤